extensions/wikia/NjordPrototype/scripts/Mom.js

```diff
@@ -0,0 +1,303 @@
+(function (window, $) {
+	'use strict';
+
+	$.fn.refresh = function () {
+		var elems = $(this.selector);
+		this.splice(0, this.length);
+		this.push.apply(this, elems);
+		return this;
+	};
+	var
+		$mwContent = $('#mw-content-text'),
+		$clonedMain = null,
+		$mainContentContainer = $('#WikiaMainContentContainer'),
+		$mainContent = $('#WikiaMainContent'),
+		$momHeader = $('#MomHeader'),
+		$editButton = $('#MomHeader .layout-edit-btn'),
+		$saveButton = $('#MomHeader .layout-save-btn'),
+		$discardButton = $('#MomHeader .layout-cancel-btn'),
+		$deleteButton = $('.mom-delete-btn'),
+		$moduleEditButton = $('.mom-edit-btn'),
+		$moduleSaveButton = $('.mom-save-btn'),
+		$moduleDiscardButton = $('.mom-discard-btn'),
+		$editMode = $('.layout-mode'),
+		$nonEditMode = $('.no-layout-mode'),
+		$momOverlays = $('.mom-module .mom-overlay'),
+		$momBar = $('.mom-module .mom-bar'),
+		$momBarContent = $('.mom-bar .mom-bar-content'),
+		$moms = $('.mom-module'),
+		$leftColumn = $('.lcs-container'),
+		$rightColumn = $('.rcs-container'),
+		initMom = function () {
+			$momHeader.show();
+			$editButton.on('click', onEdit);
+			$saveButton.on('click', onSave);
+			$discardButton.on('click', onDiscard);
+			$deleteButton.on('click', function () {
+				$(this).parents('.mom-module').remove();
+			});
+			$(window).on('scroll', onScroll);
+		}, onEdit = function () {
+			$clonedMain = $mwContent.clone(true, true);
+			$('html, body').animate({
+				scrollTop: $momHeader.offset().top - 2
+			}, 600);
+			addEmpty();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the declaration of "addEmpty" before this usage. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)